bin/smarthome.py

```diff
@@ -184,9 +185,25 @@ def __init__(self, smarthome_conf_basename=os.path.join(_etc_dir,'smarthome')):
         self.version = VERSION
         self.connections = []
 
+        if self._extern_conf_dir is not None:
+            self._etc_dir = os.path.join(self._extern_conf_dir, 'etc')
+            self._items_dir = os.path.join(self._extern_conf_dir, 'items'+os.path.sep)
+            self._logic_dir = os.path.join(self._extern_conf_dir, 'logics'+os.path.sep)
+            self._smarthome_conf_basename = os.path.join(self._etc_dir,'smarthome')
+            self._logic_conf_basename = os.path.join(self._etc_dir, 'logic')
+            self._module_conf_basename = os.path.join(self._etc_dir,'module')
+            self._plugin_conf_basename = os.path.join(self._etc_dir,'plugin')
+            self._log_config = os.path.join(self._etc_dir,'logging'+YAML_FILE)
+
+        # check config files
+        self.checkConfigFiles()
+
         # setup logging
         self.initLogging()
 
+        if self._extern_conf_dir is not None:
+            self._logger.warning("Using external config dir: {}".format(self._extern_conf_dir))
+
```

Comment:
Could always be logged (as `INFO` message) to let the user know which directory is used to lookup configuration files (in case no external directory was specified it logs the default directory - e.g. value of `_base_dir`).

Reply:
OK, since `_extern_conf_dir` will be initiallized to `_base_dir` by default, we can just always  log `_extern_conf_dir` (as `INFO`)